### ISSUE TITLE
JoinReport with unexpected results

### ISSUE THREAD

**Issue by swtrse** [mjebrahimi/BenchmarkDotNetVisualizer#11]
I do have the following Artifacts
[BenchmarkDotNet.Artifacts.zip](https://github.com/user-attachments/files/16502892/BenchmarkDotNet.Artifacts.zip)

I tried to create a joint Report
```
var options = new JoinReportHtmlOptions
              {
	              Title = "Benchmark of Collection Item Manipulation (Mean)",
	              MainColumn = "Method",
	              GroupByColumns = ["Categories", "Prefill", "PriceRange"],
	              PivotProperty = "Runtime",
	              StatisticColumns = ["Mean"],
	              ColumnsOrder = [".NET 8.0", "NativeAOT 8.0", ".NET 9.0", "NativeAOT 9.0"],
	              OtherColumnsToSelect = ["DataType"],
	              SpectrumStatisticColumn = true,
	              HighlightGroups = true,
	              DividerMode = RenderTableDividerMode.SeparateTables,
	              HtmlWrapMode = HtmlDocumentWrapMode.RichDataTables
              };

await benchmarkSummaries.JoinReportsAndSaveAsHtmlAsync(DirectoryHelper.GetPathRelativeToProjectDirectory("Reports\\Benchmark-Mean.html"), options).ConfigureAwait(false);
```
The result looks like
![image](https://github.com/user-attachments/assets/3fbf385a-9c76-4311-b9e9-e813a37a1529)
As you can see there are additional Lines (like line 2) that I did not expect. I did experiment a bit and found out that DataType maybe the culprit.

So I changed the settings to
```
var options = new JoinReportHtmlOptions
              {
	              Title = "Benchmark of Collection Item Manipulation (Mean)",
	              MainColumn = "Method",
	              GroupByColumns = ["DataType", "Categories", "Prefill", "PriceRange"],
	              PivotProperty = "Runtime",
	              StatisticColumns = ["Mean"],
	              ColumnsOrder = [".NET 8.0", "NativeAOT 8.0", ".NET 9.0", "NativeAOT 9.0"],
	              SpectrumStatisticColumn = true,
	              HighlightGroups = true,
	              DividerMode = RenderTableDividerMode.SeparateTables,
	              HtmlWrapMode = HtmlDocumentWrapMode.RichDataTables
              };

await benchmarkSummaries.JoinReportsAndSaveAsHtmlAsync(DirectoryHelper.GetPathRelativeToProjectDirectory("Reports\\Benchmark-Mean.html"), options).ConfigureAwait(false);
```
Now I get this 
![image](https://github.com/user-attachments/assets/581d3250-f4ad-4393-9a9a-d005d7dea368)
This is too fragmented, but all the doubled lines are gone and the lines are exactly what I want to show.
Basically, I want Table 1 + Table 2 joined in one table and Table 3 + Table 4 joined in one table and so on....

Did I miss some settings or maybe the Join Report do have a little flaw with merging the reports?

**Comment by swtrse** [comment]
I did provide a pull request with the fix. However I did not test for side effects but for my case this is exactly what I needed.

This is the intended result
![image](https://github.com/user-attachments/assets/3f2fe3a4-2ce3-4b54-ac62-b25a4566c3b4)

**Comment by mjebrahimi** [comment]
Thanks @swtrse. I closed this issue since you fixed it well.
